components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -117,6 +119,13 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
             if (obj != null && obj instanceof JSONArray) {
                 JSONArray redirectUris = (JSONArray) obj;
                 for (int i = 0; i < redirectUris.size(); i++) {
+                    try{
+                        URL redirectURL = new URL(redirectUris.get(i).toString());
```

Comment:
According to the relevant specifications, these are URIs. Hence it is wrong to validate them to be URLs. We have correct that to valid to be URIs.